src/components/carousel/carousel.js

```diff
@@ -92,11 +138,46 @@ class Carousel extends React.Component {
    * @method componentWillMount
    */
   componentWillMount() {
-    const selectedIndex = Number(this.props.initialSlideIndex);
+    const selectedIndex = Number(this.props.slideIndex) || Number(this.props.initialSlideIndex);
     this.setState({ selectedSlideIndex: selectedIndex });
   }
 
   /**
+   * A lifecycle method that is called before re-render.
+   *
+   * @method componentWillReceiveProps
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.slideIndex === undefined) { return; }
+
+    let newIndex = nextProps.slideIndex;
+    const currentIndex = this.state.selectedSlideIndex;
+
+    if (newIndex < 0) {
+      // If the new index is negative, select the last slide
+      newIndex = this.numOfSlides() - 1;
+    } else if (newIndex > this.numOfSlides() - 1) {
+      // If the new index is bigger than the number of slides, select the first slide
+      newIndex = 0;
+    }
+
+    if (newIndex === currentIndex) {
+      return;
+    } else if (newIndex > currentIndex) {
+      this.transitionDirection = NEXT;
```

Comment:
Is this controllable functionality? I imagine the UX of a guided, wizard-style process would want those scrolls blocked, whereas a picture gallery might not?